vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_client.go

```diff
@@ -94,6 +96,7 @@ func newContainerMetricsSnapshot(containerMetrics v1beta1.ContainerMetrics, podM
 				PodName:   podMetrics.Name,
 			},
 		},
+		PodLabels:      podMetrics.Labels,
```

Comment:
I don't think this will work. From what I see looking at the PodMetrics objects in the example cluster, they don't have the same labels as the pods have. 
See example metrics for my hamster pod (this is part of the output when running `kubectl get --raw "/apis/metrics.k8s.io/v1beta1/pods"`):
```
{
      "metadata": {
        "name": "hamster-6d7f85f5d9-r9gn8",
        "namespace": "default",
        "selfLink": "/apis/metrics.k8s.io/v1beta1/namespaces/default/pods/hamster-6d7f85f5d9-r9gn8",
        "creationTimestamp": "2019-12-03T15:50:57Z"
      },
      ...
    }
```
While `kubectl describe pod hamster-6d7f85f5d9-r9gn8` gives me
```
...
Labels:   app=hamster
          pod-template-hash=6d7f85f5d9
...
```

Reply:
I was trying out the changes now and just found out the same thing. Would it be ok if the recommender did not log warnings if there is a `KeyError` in the memory saver mode?

Reply:
Yeah, I think this is the easiest to do to not spam logs with this warning. 

Another thing is, that we could have memory saver mode query only for the metrics it actually needs , but it's out of scope of improving logging :)

Reply:
I actually considered that and ran some tests. The problem with that approach is that it's much slower than doing it this way because of how the _api-server_ gets results given a label selector. All the objects from _etcd_ are retrieved and then matched with the label selector. So multiple label selector queries would mean all the metrics would be retrieved multiple times. Much faster if it's done this way.
I will go ahead and use the second approach. Thanks.